File: languages/php/Bloock/Entity/Authenticity/HashAlg.php
<?php

namespace Bloock\Entity\Authenticity;

/**
 * Represents different hash algorithms.
 */
class HashAlg
{
    /**
     * Represents the SHA-256 hash algorithm.
     */
    const SHA_256 = "SHA_256";
    /**
     * Represents the Keccak-256 hash algorithm.
     */
    const KECCAK_256 = "KECCAK_256";
    /**
     * Represents the Poseidon hash algorithm.
     */
    const POSEIDON = "POSEIDON";
    /**
     * Represents no hash algorithm.
     */
    const NONE = "NONE";
    /**
     * Represents an unrecognized hash algorithm.
     */
    const UNRECOGNIZED = "UNRECOGNIZED";

    public static function fromProto(int $alg): string
    {
        switch ($alg) {
            case \Bloock\HashAlg::SHA_256:
                return HashAlg::SHA_256;
            case \Bloock\HashAlg::KECCAK_256:
                return HashAlg::KECCAK_256;
            case \Bloock\HashAlg::POSEIDON:
                return HashAlg::POSEIDON;
            case \Bloock\HashAlg::NONE:
                return HashAlg::NONE;
            default:
                return HashAlg::SHA_256;
        }
    }

    public static function toProto(string $type): int
    {
        switch ($type) {
            case HashAlg::SHA_256:
                return \Bloock\HashAlg::SHA_256;
            case HashAlg::KECCAK_256:
                return \Bloock\HashAlg::KECCAK_256;
            case HashAlg::POSEIDON:
                return \Bloock\HashAlg::POSEIDON;
            case HashAlg::NONE:
                return \Bloock\HashAlg::NONE;
            default:
                return \Bloock\HashAlg::SHA_256;
        }
    }
}

File: languages/python/bloock/entity/authenticity/hash_alg.py
from __future__ import annotations

from enum import Enum
import bloock._bridge.proto.authenticity_entities_pb2 as proto

class HashAlg(Enum):
    """
    Represents different hash algorithms.
    """
    SHA_256 = 0
    """
    Represents the SHA-256 hash algorithm.
    """
    KECCAK_256 = 1
    """
    Represents the Keccak-256 hash algorithm.
    """
    POSEIDON = 2
    """
    Represents the Poseidon hash algorithm.
    """
    NONE = 3
    """
    Represents no hash algorithm.
    """
    UNRECOGNIZED = -1
    """
    Represents an unrecognized hash algorithm.
    """

    def __int__(self):
        return self.value

    @staticmethod
    def from_proto(alg: proto.HashAlg.ValueType | None) -> HashAlg:
        if alg == proto.HashAlg.SHA_256:
            return HashAlg.SHA_256
        elif alg == proto.HashAlg.KECCAK_256:
            return HashAlg.KECCAK_256
        elif alg == proto.HashAlg.POSEIDON:
            return HashAlg.POSEIDON
        elif alg == proto.HashAlg.NONE:
            return HashAlg.NONE
        else:
            return HashAlg.UNRECOGNIZED

    def to_proto(self) -> proto.HashAlg:
        if self == HashAlg.SHA_256:
            return proto.HashAlg.SHA_256
        elif self == HashAlg.KECCAK_256:
            return proto.HashAlg.KECCAK_256
        elif self == HashAlg.POSEIDON:
            return proto.HashAlg.POSEIDON
        elif self == HashAlg.NONE:
            return proto.HashAlg.NONE
        else:
            return proto.HashAlg.SHA_256
